src/flattenizer.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Flattens an object
+ *
+ * @param {Object} obj - the object to unflatten
+ * @returns {Object} - the flattened object, empty if provided object is undefined
+ */
+export const flatten = (obj) => {
+   let flattened = {};
+
+    for (let key in obj) {
+        if (!obj.hasOwnProperty(key))
+            continue;
+
+        if (typeof obj[key] === 'object') {
+            let flatObject = flatten(obj[key]);
```

Comment:
Since this is the recursive step, let readers know that it's part of that by calling this `subFlattened` or something.